project-addons/custom_partner/partner.py

```diff
@@ -51,53 +54,113 @@ def _calculate_annual_invoiced(self):
         partner_ids = partner_obj.search([('is_company', '=', True), ('child_ids', '!=', False)])
         actual_year = datetime.now().year
         actual_month = datetime.now().month
+        past_month = actual_month - 1
         actual_day = datetime.now().day
         start_year = str(actual_year) + '-01-01'
+        start_past_year = str(actual_year - 1) + '-01-01'
         start_month = str(actual_year) + '-' + str(actual_month) + '-01'
+        start_past_month = str(actual_year) + '-' + str(past_month) + '-01'
         end_year = str(actual_year) + '-12-31'
+        end_past_year = str(actual_year - 1) + '-12-31'
         end_month = str(actual_year) + '-' + str(actual_month) + '-' + str(actual_day)
+        end_day_past_month = monthrange(actual_year, past_month)
+        end_past_month = str(actual_year) + '-' + str(past_month) + '-' + str(end_day_past_month[1])
         for partner in partner_ids:
             invoice_ids_year = invoice_obj.search([('date_invoice', '>=', start_year),
-                                              ('date_invoice', '<=', end_year),
-                                              '|',
-                                              ('state', '=', 'open'),
-                                              ('state', '=', 'paid'),
-                                              '|',
-                                              ('partner_id', '=', partner.id),
-                                              ('partner_id.parent_id', '=', partner.id)])
+                                                   ('date_invoice', '<=', end_year),
+                                                   '|',
+                                                   ('state', '=', 'open'),
+                                                   ('state', '=', 'paid'),
+                                                   '|',
+                                                   ('partner_id', '=', partner.id),
+                                                   ('partner_id.parent_id', '=', partner.id)])
```

Comment:
Si estáis seleccionando compañías, tenéis que buscar todas las facturas que sean del cliente e hijos, Odoo permite tener más de un nivel, en el caso de tener una factura en un segundo nivel no te la sumaría, para asegurar que coge todas podéis hacer la búsqueda con: ('partner_id', 'child_of', [partner.id]) o con ('commercial_partner_id', '=', partner_id.id) la que veais que es más eficiente. Esto se aplica a todos los search de este método, así os aseguráis que cubre todos los casos.